### DISCUSSION TITLE
Is .Net 6 subject to EAR?

### DISCUSSION THREAD

**Discussion by NPUlrk** [dotnet/runtime#74213] [Q&A] [answered]
According to [Microsoft's official website](https://www.microsoft.com/en-us/exporting/exporting-information.aspx), .NET Framework (all versions) is subject to export control. I do not find .net 6 in this list. May I ask whether .net 6 is also subject to export control?

**Comment by danmoseley** [comment]
@jamshedd  can you help?

**Reply by jamshedd** [reply]
> @jamshedd can you help?

I was expecting .NET Core would be listed, we're looking into it.

**Reply by NPUlrk** [reply]
This conclusion is important to us. If the .net 6 runtime is also subject to EAR, and given the U.S.'s increasing sanctions against Chinese companies, we have no choice but to choose another language. By the way, we all know that .net is currently open source, with developers from around the world contributing wisdom, why open source projects belong to the United States and are subject to EAR.

**Reply by MSDN-WhiteKnight** [reply]
Note that it was already discussed here: https://github.com/dotnet/csharplang/discussions/6201 The conclusion seems to be that open-source software will not be subject to restrictions.

**Reply by NPUlrk** [reply]
> 

We were relieved to see your reply and the previous [discussion](https://github.com/dotnet/csharplang/discussions/6201). Unfortunately, Microsoft updated its website to make it clear that .NET is also subject to EAR.
![image](https://user-images.githubusercontent.com/21106497/187819967-e3f10b35-c002-46bc-baa1-6e326c7e0aba.png)

**Reply by NPUlrk** [reply]
Microsoft once lost a lot of .net developers because of closed source and bound Windows platform. Now, will it lose Chinese developers again because of cooperating with the US government to curb China's development?

**Reply by MSDN-WhiteKnight** [reply]
The inclusion to this list might not matter much, though. For example, Microsoft stopped shipping all products in Russia, but downloading .NET from Russia is not blocked. Even Windows updates are not blocked. It seems these restrictions are just for show and Microsoft is not really interested in "curbing" anyone. Or do you need this for compliance reasons, i.e. you are required to only use technologies that are legal in China?

**Reply by NPUlrk** [reply]
> 

Yes, our company requires software to be used in compliance with laws (including US export controls). As we all know, in terms of semiconductor manufacturing, the United States currently does not allow third-party companies that manufacture for Huawei to use any products containing American technology. We are not sure whether C# and .net will be subject to the same restrictions.

**Reply by huoyaoyuan** [reply]
Open source software is somehow different from others. With source-build infrastructures, ones can always build their own versions, as long as they can access any clone of the repository. The license should allow this too.

How does your policy apply to Linux-distrubuted builds? For example, .NET on RHEL is "supported by RedHat".

**Reply by NPUlrk** [reply]
@huoyaoyuan Using technology that is added to the EAR list is always a risk for companies that comply strictly with the law. My company is a well-known multinational company. Many teams used C# and .net technologies, but they were forced to switch to Java or Python since the US sanctions. Experts with a voice say that C# and .Net belong to Microsoft and will affect the company's business continuity. We've been facing questions from all sides about the continued use of .Net. My intention is to get evidence here that .net is not controlled by EAR and Microsoft, but unfortunately I can only tell them that there are risks and that we may be able to compile .net ourselves once restricted. I hope this day doesn't come. I want technology to be truly borderless, and C# and net can benefit everyone, every corner of the planet.

**Reply by Clockwork-Muse** [reply]
Given [this Linux Foundation whitepaper](https://www.linuxfoundation.org/tools/understanding-us-export-controls-with-open-source-projects/), .NET is probably over-classified.  

Or it may not outright prevent use in China.  Consider that Azure is available in China

**Reply by MSDN-WhiteKnight** [reply]
Java is maintained by Oracle, also a company registered in USA. I hope you write "Experts with a voice" in ironical sence? If they suggest to replace .NET with Java with a goal to get rid of american technologies, they obviously don't know what they are talking about.

**Reply by NPUlrk** [reply]
@Clockwork-Muse It would be best if the Microsoft .net team or the .net foundation could make it clear that .net is not subject to the EAR as the Linux Foundation has done. By our own interpretation, 1000 people have a thousand Hamlets.

**Reply by NPUlrk** [reply]
@MSDN-WhiteKnight They think that the java language is open and unrestricted. As for the JDK, they have developed their own JDK, so there is no risk. The most important thing is that they are still in the era when .net is closed source and belongs to Microsoft.

**Reply by jamshedd** [reply]
We are working with our export controls team to get clarity on this issue and will post an update when there's something to share.

**Comment by jamshedd** [comment, marked as answer]
Hi folks, I have an update here - after discussions with our trade team and legal we determined we can classify all versions of .NET/.NET Core as "Not subject to EAR". This change is now reflected on the Export Control website: https://www.microsoft.com/en-us/exporting/exporting-information.aspx

_Note: .NET **Framework** unlike modern .NET (NET Core) is a component of Windows and closed source, therefore all versions of .NET Framework will continue to be classified 5D992.c and are subject to EAR_

**Reply by NPUlrk** [reply]
It's a very exciting decision.

**Reply by GFLEE** [reply]
"Nobody knows more about programs than me.  I know more about programming than any human being on Earth. , I have been programming for 40 years since i was born, the chinese should use their own frameworks. They are take advantage of us"

**Reply by hez2010** [reply]
Thanks for the clarification. With this clarification those who are affected by EAR can use .NET without any concerns.

**Reply by flyingfive** [reply]
> "Nobody knows more about programs than me. I know more about programming than any human being on Earth. , I have been programming for 40 years since i was born, the chinese should use their own frameworks. They are take advantage of us"

amazing!

**Comment by EminemJK** [comment]
https://www.microsoft.com/en-us/exporting/exporting-information

**Comment by jamshedd** [comment]
This discussion is closed, the issue is already resolved with the Microsoft export controls website clearly stating modern .NET i.e. .NET Core is not subject to EAR.

![image](https://github.com/dotnet/runtime/assets/9440640/4e2e36d1-4705-48b8-9875-86d19c5e701c)
